src/components/utils/hooks/useXmtpClient.tsx

```diff
@@ -29,15 +29,17 @@ const wipeKeys = (walletAddress: string) => {
 };
 
 const useXmtpClient = () => {
-  const { data: signer } = useSigner();
+  const { data: signer, isLoading } = useSigner();
   const currentProfile = useAppStore((state) => state.currentProfile);
   const client = useMessageStore((state) => state.client);
   const setClient = useMessageStore((state) => state.setClient);
+  const [awaitingXmtpAuth, setAwaitingXmtpAuth] = useState<boolean>();
   const isMessagesEnabled = isFeatureEnabled('messages', currentProfile?.id);
 
   useEffect(() => {
     const initXmtpClient = async () => {
       if (signer && !client && currentProfile) {
+        setAwaitingXmtpAuth(true);
```

Comment:
Think we want this in the `if (!keys) {` block. Otherwise it is going to set the `awaitingXmtpAuth` to true even when the keys are cached in LocalStorage

Reply:
Good catch! Updated 👍